Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -736,10 +736,23 @@ public struct RenderNodeTranslator: SemanticVisitor {
         node.topicSectionsStyle = topicsSectionStyle(for: documentationNode)
         
         if shouldCreateAutomaticRoleHeading(for: documentationNode) {
+            // If there are no links to other nodes from the article,
+            // set the eyebrow for articles.
             if node.topicSections.isEmpty {
                 // Set an eyebrow for articles
                 node.metadata.roleHeading = "Article"
             }
+            // If the article links to other nodes, set the eyebrow for
+            // API Collections if any linked node is a symbol.
+            else {
+                let isAPICollection = contentCompiler.collectedTopicReferences.contains { topicReference in
+                    context.topicGraph.nodeWithReference(topicReference)?.kind.isSymbol ?? false
+                }
+                if isAPICollection {
+                    // Set an eyebrow for API Collection
+                    node.metadata.roleHeading = "API Collection"
+                }
+            }
             node.metadata.role = DocumentationContentRenderer.roleForArticle(article, nodeKind: documentationNode.kind).rawValue
```

Comment:
Now that this role heading code is more than one check I think it could make sense to leverage the logic that's already there to determine the role. That way the only unique check here is if the page curates any symbols.

I thought that the tests would fail if I added a symbol link in the abstract of the collection page but was surprised to see that `contentCompiler` is only used for the topic sections (and see also sections) and that the abstract and discussion rendering create their own content compilers. If we're relying on `contentCompiler` only being used for topic section content then I feel that it would be good to rename it to `topicSectionsContentCompiler` and to move it down to shortly before its first used. It would also be good if the test had a symbol link in the content of the collection page to catch if the `topicSectionsContentCompiler` was ever used for other content on the page.

```suggestion
            let role = DocumentationContentRenderer.roleForArticle(article, nodeKind: documentationNode.kind)
            node.metadata.role = role.rawValue
            
            switch role {
            case .article:
                node.metadata.roleHeading = "Article"
            case .collectionGroup:
                let curatesSymbols = topicSectionsContentCompiler.collectedTopicReferences.contains { topicReference in
                    context.topicGraph.nodeWithReference(topicReference)?.kind.isSymbol ?? false
                }
                if curatesSymbols {
                    node.metadata.roleHeading = "API Collection"
                }
            default:
                break
            }
```

Reply:
> If we're relying on contentCompiler only being used for topic section content then I feel that it would be good to rename it to topicSectionsContentCompiler and to move it down to shortly before its first used.

I believe that keeping it as `contentCompiler`is the best option for now. Even though, in the case of `RendernodeTranslator` it is only used to store references to topics, it could also be utilized for abstracts, images, groups, and more.

If we aim for consistency and decide to refactor `contentCompiler` to `topicSectionsContentCompiler` we would also need to update all other declarations of `contentCompiler` to their corresponding specific names. For example, in the `Snippet` file, we declare `contentCompiler` and only use it for snippets. In this case, it should be renamed to `snippetsContentCompiler` or something along those lines. This renaming process would apply to many other files as well.

I believe that such a refactor would be better to scope it in its own pull request.

Reply:
Thanks for the feedback David, I addressed the comments in this commit https://github.com/apple/swift-docc/pull/754/commits/8f240df5c6b826e8ad2901622bdedbe8458b4acd

Reply:
> I believe that keeping it as `contentCompiler` the best option for now. Even though, in the case of `RenderNodeTranslator` it is only used to store references to topics, it could also be utilized for abstracts, images, groups, and more.

In that case please update the tests to be able to catch any regressions if someone used the `contentCompiler` for other content. Adding abstracts and discussion sections with symbol links to the generated test markdown and abstracts with symbol links in the topic sections. For example;

```diff
 try """
 # Collection
+ An abstract with a symbol link: ``MyKit/MyProtocol`` 
 
+ An overview with a symbol link: ``MyKit/MyProtocol`` 

 ## Topics
+ A topic group abstract with a symbol link: ``MyKit/MyProtocol`` 
 
 - <doc:article>
 """.write(to: url.appendingPathComponent("article.md"), atomically: true, encoding: .utf8)
```

Reply:
`contentCompiler` is a local variable.

Reply:
Yes, but it's local to the entire 280 line `visitArticle` function and the correctness of this code relies on the content compiler only being used for the topic sections (since `contentCompiler.collectedTopicReferences` is used to get all the curated references). 

Because it would introduce a bug if the `contentCompiler` was used for any other content in the scope that it is local and available to; we should either rename it to indicate that it should only be used for topic sections or add tests that verify that symbol references in the rest of the articles content aren't considered when checking if the curation included any symbols.

Reply:
Renamed to `topicSectionContentCompiler`here https://github.com/apple/swift-docc/pull/754/commits/bff738ea96ee4a1daebb83ddbafa91d482683b9d